airbyte/_util/meta.py

```diff
@@ -32,6 +32,39 @@ def is_ci() -> bool:
     return "CI" in os.environ
 
 
+_MCP_MODE_ENABLED: bool = False
+
+
+def _set_mcp_mode(*, enabled: bool) -> None:
+    """Set MCP mode flag. This is a private function - use enable_mcp_mode() instead."""
+    global _MCP_MODE_ENABLED
+    _MCP_MODE_ENABLED = enabled
+
+
+def enable_mcp_mode() -> None:
+    """Enable MCP (Model Context Protocol) mode.
+
+    This should be called early in MCP server initialization to ensure
+    proper detection and prevent interactive prompts.
+    """
+    _set_mcp_mode(enabled=True)
+
+
+def is_mcp() -> bool:
+    """Return True if running in MCP (Model Context Protocol) mode."""
+    if _MCP_MODE_ENABLED:
+        return True
+
+    script_name = get_python_script_name()
+    if script_name and "airbyte-mcp" in script_name:
+        return True
+
+    if "fastmcp" in sys.modules:
+        return True
+
+    return "MCP_SERVER" in os.environ or "AIRBYTE_MCP_MODE" in os.environ
```

Comment:
These last two conditions could create false positives. Just remove these and return false if we reach here.